### ISSUE TITLE
False positives for `Naming/BlockForwarding` where the block has a fallback

### ISSUE THREAD

**Issue by benedikt** [rubocop/rubocop#10465]
The `Naming/BlockForwarding` cop is returning offenses for methods where the named block variable has a fallback defined via `||=`. Autofixing this changes the methods behavior. 

```ruby
def example(&block)
  block ||= -> { 'something' }
  [].map(&block)
end
```

--------

## Expected behavior

No offenses detected.

## Actual behavior

```
$ rubocop --debug 
For /Users/benedikt/Workspace/Ruby/rubocop-example: configuration from /Users/benedikt/Workspace/Ruby/rubocop-example/.rubocop.yml
Default configuration from /Users/benedikt/.rbenv/versions/3.1.1/lib/ruby/gems/3.1.0/gems/rubocop-1.26.1/config/default.yml
Use parallel by default.
Running parallel inspection
Loading cache from /Users/benedikt/.cache/rubocop_cache/c214916517681acc71076fdbd8cd5e24c8caa2a6/6d7a3b621ca1730e04accd938619e4bdab66cfb1/8b686c07dc716bd5a417d5214ae1fd0143f01685
Inspecting 1 file
Scanning /Users/benedikt/Workspace/Ruby/rubocop-example/example.rb
Loading cache from /Users/benedikt/.cache/rubocop_cache/c214916517681acc71076fdbd8cd5e24c8caa2a6/6d7a3b621ca1730e04accd938619e4bdab66cfb1/8b686c07dc716bd5a417d5214ae1fd0143f01685
C

Offenses:

example.rb:3:13: C: [Correctable] Naming/BlockForwarding: Use anonymous block forwarding.
def example(&block)
            ^^^^^^
example.rb:5:10: C: [Correctable] Naming/BlockForwarding: Use anonymous block forwarding.
  [].map(&block)
         ^^^^^^

1 file inspected, 2 offenses detected, 2 offenses auto-correctable
Finished in 0.15310699999099597 seconds
```

## Steps to reproduce the problem

This only happens on Ruby 3.1, and I'm using this basic `.rubocop.yml`: 

```yaml
AllCops:
  TargetRubyVersion: 3.1
  NewCops: enable
```

Here's a minimal `example.rb` file that shows the problem:

```ruby
# frozen_string_literal: true

def example(&block)
  block ||= -> { 'something' }
  [].map(&block)
end
```

## RuboCop version

Include the output of `rubocop -V` or `bundle exec rubocop -V` if using Bundler.
If you see extension cop versions (e.g. `rubocop-performance`, `rubocop-rspec`, and others)
output by `rubocop -V`, include them as well. Here's an example:

```
$ rubocop -V
1.26.1 (using Parser 3.1.1.0, rubocop-ast 1.16.0, running on ruby 3.1.1 x86_64-darwin21)
```

**Comment by benedikt** [comment]
Thanks for the quick fix!
